plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/dependency/DependencyCollector.java

```diff
@@ -42,70 +45,72 @@ public class DependencyCollector {
 
   private static final String UNKNOWN = "unknown";
 
-  private final Project project;
-  private final Set<File> exclusionFromDependencies;
-  private final Set<GradleModuleDependency> moduleDependencies;
-
-  /**
-   * Instantiates a new dependency collector.
-   */
-  public DependencyCollector(Project project, Set<File> exclusionFromDependencies) {
-    this.project = project;
-    this.exclusionFromDependencies = exclusionFromDependencies;
-    this.moduleDependencies = new LinkedHashSet<>();
-  }
-
-  public Set<GradleModuleDependency> getModuleDependencies() {
-    return moduleDependencies;
-  }
-
   /**
    * Resolve and collect dependencies from a {@link SourceSet}.
    */
-  public void collectByConfigurationNames(Set<String> configurationNames) {
-    if (GradleVersion.current().compareTo(GradleVersion.version("3.3")) < 0) {
-      List<ResolvedConfiguration> configs = project.getConfigurations().stream()
-              .filter(configuration -> configurationNames.contains(configuration.getName()))
+  public static Set<GradleModuleDependency> getModuleDependencies(Project project,
+      Set<String> configurationNames) {
+    if (GradleVersion.current().compareTo(GradleVersion.version("4.0")) < 0) {
```

Comment:
Is this due to the `ArtifactView.ViewConfiguration` compatibility?

Reply:
Yes - when I added this in the previous PR I put `3.3` as the version when Gradle started to support it but actually it should be `4.0`.
The code paths haven't really changed in `DependencyCollector`, I've just switched it from populating the dependencies into the source set to returning a list of dependencies.
Then the exclusions are removed later in the build action.